# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/cairo-device.c
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/cairo-device.c
--- a/gfx/cairo/cairo/src/cairo-device.c	Mon May 06 19:22:35 2024 +0000
+++ b/gfx/cairo/cairo/src/cairo-device.c	Mon May 06 19:25:12 2024 +0000
@@ -163,6 +163,8 @@
     case CAIRO_STATUS_FREETYPE_ERROR:
     case CAIRO_STATUS_WIN32_GDI_ERROR:
     case CAIRO_STATUS_TAG_ERROR:
+    case CAIRO_STATUS_DWRITE_ERROR:
+    case CAIRO_STATUS_SVG_FONT_ERROR:
     default:
 	_cairo_error_throw (CAIRO_STATUS_NO_MEMORY);
 	return (cairo_device_t *) &_nil_device;
@@ -215,7 +217,6 @@
 
     return device;
 }
-slim_hidden_def (cairo_device_reference);
 
 /**
  * cairo_device_status:
@@ -270,7 +271,6 @@
 	    status = _cairo_device_set_error (device, status);
     }
 }
-slim_hidden_def (cairo_device_flush);
 
 /**
  * cairo_device_finish:
@@ -314,7 +314,6 @@
      */
     device->finished = TRUE;
 }
-slim_hidden_def (cairo_device_finish);
 
 /**
  * cairo_device_destroy:
@@ -355,7 +354,6 @@
     _cairo_user_data_array_fini (&user_data);
 
 }
-slim_hidden_def (cairo_device_destroy);
 
 /**
  * cairo_device_get_type:
@@ -430,7 +428,6 @@
 
     return CAIRO_STATUS_SUCCESS;
 }
-slim_hidden_def (cairo_device_acquire);
 
 /**
  * cairo_device_release:
@@ -456,7 +453,6 @@
 
     CAIRO_MUTEX_UNLOCK (device->mutex);
 }
-slim_hidden_def (cairo_device_release);
 
 cairo_status_t
 _cairo_device_set_error (cairo_device_t *device,